### PR TITLE
[Recharts] Add payload in Dot Props for Area and Line components

### PR DESCRIPTION
**Pull request by gndelia**
Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] Add or edit tests to reflect the change. (Run with `npm test`.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:

- [x] Provide a URL to documentation or source code which provides context for the suggested changes:  http://recharts.org/en-US/api/Line and the official example linked https://jsfiddle.net/alidingling/9y9zrpjp/
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [x] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

File: types/recharts/index.d.ts
Comment (gndelia) [inline types/recharts/index.d.ts:30]: these were unused, so I'm removing them

**Comment by danger-public** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## recharts ([<kbd>unpkg</kbd>](https://unpkg.com/browse/recharts@latest/))
was missing the following properties:
1. Layer
1. Sankey
1. Funnel
1. FunnelChart
1. Trapezoid
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against e1432544285f1a42a0526d3aafb21210b61f3dd6
</p>

**Comment by elibarzilay** [comment]
@gndelia: is this ready, or should it wait for a review? (Ping @rapmue @royxue @ZheyangSong @richbai90 @caspeco-dan @pkeuter @jrsaunde @CruseCtrl @apalugniok @RobertStigsson @kousaku-maron @iflp @veddermatic @G07cha @dimitropoulos @eliotball @vkentta @fcaylus @samuelweckstrom @Gerhut @haldunanil @t-knapp)

**Comment by gndelia** [comment]
it's ready for review - I'm just waiting for any approval

**Review by G07cha — APPROVED** [review]

**Comment by gndelia** [comment]
Ready to merge

**Pull request merged**
